subprojects/composite-builds/src/main/java/org/gradle/composite/internal/plugins/LocalPluginResolution.java

```diff
@@ -40,6 +40,11 @@ public PluginId getPluginId() {
         return pluginId;
     }
 
+    @Override
+    public String getPluginVersion() {
+        return producingProject.getVersion().toString();
```

Comment:
If I understand correctly, this takes care of the case where the plugin is produced by a project (included build or not).
I wonder what we should do in that case. Effectively having a version declared means little since you want to use a "live" version anyway.
Should we special case this more? Not very much in favor
Or instead fail on mismatch, assuming that using a version catalog for this setup is not frequent

Reply:
Either I can remove this return value (not sure if that prevents anything, since those would be declared in the settings file anyways), or I could add some sort of "source identifier" that is compared alongside the version. The former seems like a better idea to me.

Reply:
Had some discussion on this point - let's leave it as-is for now. If we end up having requests for that specific corner case, it can be added later.